src/sharing/SharingScreen.js

```diff
@@ -50,51 +47,45 @@ const styles = createStyleSheet({
   },
 });
 
-class SharingScreen extends PureComponent<Props> {
-  render() {
-    const { auth } = this.props;
-    const { params } = this.props.route;
-
-    // If there is no active logged-in account, abandon the sharing attempt,
-    // and present the account picker screen to the user.
-    if (auth === undefined) {
-      NavigationService.dispatch(navigateToAccountPicker());
-      return null;
-    }
+export default function SharingScreen(props: Props) {
+  const { params } = props.route;
+  const auth = useSelector(tryGetAuth);
```

Comment:
Looking at this, I see that we're not actually using it for more than its truthiness value. So we could use `getHasAuth` instead, which would be a bit more transparent about the purpose and also make one fewer place there's an `Auth` value floating around.

Reply:
Makes sense! #4878.